control-plane/agents/src/bin/core/controller/registry.rs

```diff
@@ -186,9 +193,26 @@ impl Registry {
         config
     }
 
+    /// Disable the v1 compat mode, once all nexus info keys are migrated.
+    pub(crate) async fn disable_v1_compat<S: Store>(store: &mut S) -> CoreRegistryConfig {
+        let config = CoreRegistryConfig::new(NodeRegistration::Automatic);
+        store
+            .put_obj(&config)
+            .await
+            .expect("Failed to persist configuration information");
+        info!("v1 compatibility is disabled");
+        config
+    }
+
     /// Get the `CoreRegistryConfig`
-    pub(crate) fn config(&self) -> &CoreRegistryConfig {
-        &self.config
+    pub(crate) fn config(&self) -> RwLockReadGuard<'_, RawRwLock, CoreRegistryConfig> {
```

Comment:
```suggestion
    pub(crate) fn config(&self) -> parking_lot::RwLockReadGuard<CoreRegistryConfig> {
```
This should suffice..